src/test/librados/test_common.cc

```diff
@@ -132,7 +132,6 @@ std::string with_healthy_cluster(rados_t* cluster,
       throw pool_op_error{pool_name, "wait_for_healthy", r};
     }
   } catch (const pool_op_error& e) {
-    rados_shutdown(*cluster);
```

Comment:
also, why do you think this fixes https://tracker.ceph.com/issues/42706 ? how is this call related to the return value of `with_healthy_cluster()`?

Reply:
set_pg_num will call with_healthy_cluster, and if set_pg_num failed, it will raise pool_op_error,
that will call rados_shutdown, and ASSERT_TRUE(set_pg_num(&s_cluster, pool_name, 11).empty());
will fail, then will call RadosTestNS::TearDown(), but in this test suite it will clean the objects first, then call RadosTestNS::cleanup_all_objects().

Reply:
if a `pool_op_error` is thrown, `set_pg_num()` will return a non-empty string. so the test still fails with your change. am i missing something?

Reply:
it's ok to fail, the issue is it shouldn't segfault.

Reply:
i appreciate this fix. but could you please remove "Fixes: https://tracker.ceph.com/issues/42706" from the commit message. i still don't think this change addresses the failure describe by it.